src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1219,6 +1232,28 @@ private HttpResponseHeaders CreateResponseHeaders(bool appCompleted)
         return responseHeaders;
     }
 
+    private bool CanIncludeResponseContentLengthHeader()
+    {
+        // Section 4.3.6 of RFC7231
+        if (Is1xxCode(StatusCode) || StatusCode == StatusCodes.Status204NoContent)
+        {
+            // A server MUST NOT send a Content-Length header field in any response
+            // with a status code of 1xx (Informational) or 204 (No Content).
+            return false;
+        }
+        else if (Method == HttpMethod.Connect && Is2xxCode(StatusCode))
+        {
+            // A server MUST NOT send a Content-Length header field in any 2xx
+            // (Successful) response to a CONNECT request.
+            return false;
+        }
+
+        return true;
+
+        static bool Is1xxCode(int code) => code >= StatusCodes.Status100Continue && code < StatusCodes.Status200OK;
+        static bool Is2xxCode(int code) => code >= StatusCodes.Status200OK && code < StatusCodes.Status300MultipleChoices;
```

Comment:
```suggestion
        static bool Is1xxCode(int code) => code is >= StatusCodes.Status100Continue and < StatusCodes.Status200OK;
        static bool Is2xxCode(int code) => code is >= StatusCodes.Status200OK and < StatusCodes.Status300MultipleChoices;
```
Cf. https://github.com/dotnet/roslyn/issues/60534 -- Roslyn can produce optimized code with the pattern matching by using an `uint`-cast that saves one comparison at runtime. Hopefully this will be implemented soon 😉